src/tools/rest-backend/database_application.cpp

```diff
@@ -0,0 +1,902 @@
+			data["value"][j]["plugin"] = elem.getPluginformat ().getPluginname ();
+			data["value"][j]["value"] = elem.getConfig ();
+			data["value"][j]["validated"] = elem.isValidated ();
+			j++;
+		}
+
+		RootApp::setOk (resp, data, "application/json");
+	}
+	catch (exception::EntryNotFoundException & e)
+	{
+		RootApp::setNotFound (resp, "The requested entry does not exist.", "ENTRY_DOES_NOT_EXIST");
+		return;
+	}
+}
+
+/**
+ * @brief handles a creation request for a new snippet entry.
+ * @param req a request
+ * @param resp a response
+ */
+inline void DatabaseApp::handleInsert (cppcms::http::request & req, cppcms::http::response & resp)
+{
+	// first check if the currently authenticated user may create new database entries
+	if (!AuthenticationApp::validateAuthentication (req, resp, ELEKTRA_REST_PERMISSIONS_CREATE_ENTRY))
+	{
+		return; // quit early, error message set already
+	}
+
+	model::User currentUser ("");
+	try
+	{
+		currentUser = AuthenticationApp::getCurrentUser (req);
+	}
+	catch (exception::ElektraRestException & e)
+	{
+		RootApp::setUnauthorized (resp, "You need to be authenticated to perform this action.", "NEED_AUTHENTICATION");
+		return; // quit early
+	}
+
+	model::Entry entry ("");
+	try
+	{
+		entry = this->buildAndValidateEntry (req, resp);
+	}
+	catch (exception::EntryValidationException & e)
+	{
+		// error already written to output
+		return; // so quit early
+	}
+
+	if (service::StorageEngine::instance ().entryExists (entry.getPublicName ()))
+	{
+		RootApp::setBadRequest (resp, "An entry with the given details does already exist.", "ENTRY_DOES_ALREADY_EXIST");
+		return; // quit early
+	}
+
+	try
+	{
+		service::StorageEngine::instance ().createEntry (entry);
+	}
+	catch (exception::EntryAlreadyExistsException & e)
+	{
+		RootApp::setBadRequest (resp, "An entry with the given details does already exist.", "ENTRY_DOES_ALREADY_EXIST");
+		return; // quit early
+	}
+
+	RootApp::setOk (resp, "The entry has been created successfully.", "ENTRY_CREATED_SUCCESSFULLY");
+}
+
+/**
+ * @brief handles an update request for a snippet entry.
+ * @param req a request
+ * @param resp a response
+ * @param key the unique snippet key of the entry which shall be updated
+ */
+inline void DatabaseApp::handleUpdate (cppcms::http::request & req, cppcms::http::response & resp, std::string & key)
+{
+	// first lets find the entry
+	model::Entry oldEntry ("");
+	try
+	{
+		oldEntry = service::StorageEngine::instance ().getEntry (key);
+	}
+	catch (exception::EntryNotFoundException & e)
+	{
+		RootApp::setNotFound (resp, "The entry to update does not exist.", "ENTRY_DOES_NOT_EXIST");
+		return; // quit early
+	}
+
+	// now check if the currently authenticated user may even update it
+	if (!AuthenticationApp::validateAuthentication (req, resp, ELEKTRA_REST_PERMISSIONS_UPDATE_ENTRY, oldEntry.getAuthor ()))
+	{
+		return; // quit early, error response already set
+	}
+
+	model::Entry newEntry ("");
```

Comment:
This one is actually only 50 lines long, while our coding standard is fine up to 100 lines?!